cas-server-webapp/src/main/resources/messages.properties

```diff
@@ -94,6 +94,7 @@ INVALID_TICKET=Ticket ''{0}'' not recognized
 INVALID_SERVICE=Ticket ''{0}'' does not match supplied service. The original service was ''{1}'' and the supplied service was ''{2}''.
 INVALID_PROXY_CALLBACK=The supplied proxy callback url ''{0}'' could not be authenticated.
 UNAUTHORIZED_SERVICE_PROXY=The supplied service ''{0}'' is not authorized to use CAS proxy authentication.
+UNSATISFIED_AUTHN_POLICY=Service access denied due to an unsatisfied authentication policy.
```

Comment:
it was the pre-existing error code for `UnsatisfiedAuthenticationPolicyException` https://github.com/frett/cas/blob/ticketExceptions/cas-server-core-authentication/src/main/java/org/jasig/cas/ticket/UnsatisfiedAuthenticationPolicyException.java#L19

This is one of the points that handling of `AbstractTicketException`s diverged from handling of `AuthenticationException`s. For `AuthenticationException` we don't have access to the original underlying exception thrown by the auth handler, we only have access to the class. So, we need to generate a message code based off of that class.

For `AbstractTicketException` we have access to the actual exception thrown, so we can utilize the built-in `getCode()` method to resolve which error message to show. I opted to go this route over generating a fixed error code based on the exception class name so that it is possible to dynamically switch out the error message when throwing the original `AbstractTicketException`.